Server/VikingVault.Services/UserService.cs

```diff
@@ -13,23 +13,63 @@ namespace VikingVault.Services
     public class UserService : IUserService
     {
         private readonly VikingVaultDbContext _dbContext;
-        public UserService(VikingVaultDbContext dbContext)
+        private readonly IBankAccountService _bankAccountService;
+
+        public UserService(VikingVaultDbContext dbContext, IBankAccountService bankAccountService)
         {
             _dbContext = dbContext;
+            _bankAccountService = bankAccountService;
         }
 
         public User CreateUser(User user)
         {
             user.Role = "user";
+            user.Password = PasswordEncryption.ComputeSha256Hash(user.Password);
+
+            BankAccount ronBankAccount = new BankAccount
```

Comment:
This chunks of code can be extracted to methods "CreateBankAccount" that get some parameters. That way, we would reuse some code and everything would be smoother and easier to read.